Rocket.Chat/Extensions/UIBarButtonItem+Badge.swift

```diff
@@ -0,0 +1,142 @@
+//
+//  UIBarButtonItem+Badge.swift
+//  Rocket.Chat
+//
+//  Created by Lucas Woo on 4/26/17.
+//  Copyright © 2017 Rocket.Chat. All rights reserved.
+//
+
+import Foundation
+import ObjectiveC
```

Comment:
@legendecas Well, my idea is to have a custom `UIView` subclass that will have all the properties and subviews related to the badge. I think it's much more clear and easier to write tests. Can you give it a try?